cmd/kube-controller-manager/app/options/options_test.go

```diff
@@ -657,6 +656,96 @@ func TestApplyTo(t *testing.T) {
 	}
 }
 
+func TestEmulatedVersion(t *testing.T) {
+	var cleanupAndSetupFunc = func() featuregate.FeatureGate {
+		componentGlobalsRegistry := utilversion.DefaultComponentGlobalsRegistry
+		componentGlobalsRegistry.Reset() // make sure this test have a clean state
+		t.Cleanup(func() {
+			componentGlobalsRegistry.Reset() // make sure this test doesn't leak a dirty state
+		})
+
+		verKube := utilversion.NewEffectiveVersion("1.32")
+		fg := featuregate.NewVersionedFeatureGate(version.MustParse("1.32"))
+		utilruntime.Must(fg.AddVersioned(map[featuregate.Feature]featuregate.VersionedSpecs{
+			"kubeA": {
+				{Version: version.MustParse("1.32"), Default: true, LockToDefault: true, PreRelease: featuregate.GA},
+				{Version: version.MustParse("1.30"), Default: false, PreRelease: featuregate.Beta},
+			},
+			"kubeB": {
+				{Version: version.MustParse("1.31"), Default: false, PreRelease: featuregate.Alpha},
+			},
+		}))
+		utilruntime.Must(componentGlobalsRegistry.Register(utilversion.DefaultKubeComponent, verKube, fg))
+		return fg
```

Comment:
Does this test use versions surrounding the current version at the time the test was written because it won't work when the current version is different?  Will this test break in future releases? 

cc @siyuanfoundation

Reply:
No, it won't break in future releases because `utilversion.NewEffectiveVersion("1.32")` fixed the binary version of the test to 1.32